moses/moses/main/populate_atomspace.cc

```diff
@@ -0,0 +1,62 @@
+#include <moses/data/table/table.h>
+#include <moses/data/table/table_io.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <moses/utils/value_key.h>
+#include <opencog/atoms/base/Link.h>
+#include "populate_atomspace.h"
+#include <opencog/atoms/value/Value.h>
+
+namespace opencog { namespace moses {
+
+void populate(AtomSpace *as, ITable &itable) {
+
+	std::vector<multi_type_seq>::const_iterator it, end;
+	int col_size = itable.get_types().size();
+	for (int i = 0; i < col_size; i++) {
+		id::type_node col_type = itable.get_types().at(i);
+		int row_size = itable.size();
+		vertex_seq col = itable.get_column_data(i);
+		switch (col_type) {
+			case id::boolean_type: {
+				std::vector<ValuePtr> col_values = {};
+				Handle feature = createNode(SCHEMA_NODE, itable.get_labels().at(i));
+				for (int j = 0; j < row_size; j++) {
+					// for each element the ith column and the jth row
+					// change the vertex to bool
+					bool col_data = vertex_to_bool(col.at(j));
+					col_values.push_back(ValuePtr(createLink(col_data ? TRUE_LINK : FALSE_LINK)));
+				}
+				ValuePtr  vtr(new LinkValue(col_values));
+				feature->setValue(value_key, vtr);
+				as->add_atom(feature);
+				break;
+			}
+			case id::contin_type: {
+				std::vector<double> col_values_contin = {};
+				Handle feature = createNode(SCHEMA_NODE, itable.get_labels().at(i));
+				for (int j = 0; j < row_size; j++)
+					col_values_contin.push_back(
+							get_contin(col.at(j)));
+
+				ValuePtr vtr(new FloatValue(col_values_contin));
+				feature->setValue(value_key, vtr);
+				as->add_atom(feature);
+				break;
+			}
+			case id::enum_type: {
+				//TODO enum_type data to be added to Atomspace
+				break;
+			}
+			default: {
+				std::stringstream ss;
+				ss << col_type;
+				throw ComboException(TRACE_INFO,
+				                     "populate atomspace can not handle type_node %s",
+				                     ss.str().c_str());
+			}
+
+		}
+	}
+}
```

Comment:
The code is good. Although ideally I would offer you to move the code that turns a `vertex_seq` into a `ValuePtr` in its own function. First, it would make reading that code easier. Second, it will remove redundant lines like
```c++
			feature->setValue(value_key, vtr);
			as->add_atom(feature); 
```
as you'll only need that once. And third, it will make it easier to grow in the future.

So that function would have the signature
```c++
ValuePtr vertex_seq_to_value(const vertex_seq&, id::type_node)
```

See what I mean?

Reply:
I see your point but why is it necessary to pass node types to the vertex_seq_to_value

Reply:
here is how i thought of it
`ValuePtr vertex_seq_to_value(vertex_seq& col, int rsize)`

Reply:
@Eman22S `vertex_seq_to_value` would work for any type, not just contin. So it would include the switch/case statement.

Reply:
So are you suggesting atomspace population should be different from vertex_seq to value conversion or simply move the switch/case statement to the vertex_seq function. Where should add atoms occur??

Reply:
i made code responsible for atomspace population in populate(atom addition) and vertex_seq to value conversion in `vertex_seq_to_value`. Was that your suggestion ??

Reply:
@Eman22S I don't find any `vertex_seq_to_value` function. What I suggest is to have the switch/case statement in the `vertex_seq_to_value` but instead of attaching `vtr` to `feature` and adding `feature` to `as` it would just return `vtr`, then attaching the result of `vertex_seq_to_value` to `feature` and adding to `as` would occur inside `populate`.

Reply:
I haven't  commited it yet.

Reply:
Good remark. It could indeed be in another file, especially if it happens to be useful elsewhere (I can't think of whether that's the case). Or perhaps a method of `ITable` like `ITable::get_column_value` which would have `populate` directly call it instead of `ITable::get_column_data`.

But it's up to you whether you think it is worthwhile at that point, I'm neutral about it.